smt.go

```diff
@@ -423,27 +418,24 @@ func (smt *SparseMerkleTree) doProveForRoot(key []byte, root []byte, isUpdatable
 	// value, we do not need to add anything else to the proof.
 	var nonMembershipLeafData []byte
 	if !bytes.Equal(pathNodes[0], smt.th.placeholder()) {
-		actualPath, _ := smt.th.parseLeaf(leafData)
-		if !bytes.Equal(actualPath, path) {
+		actualKey, _ := smt.th.parseLeaf(leafData, smt.values.GetKeySize())
+		if !bytes.Equal(actualKey, key) {
 			// This is a non-membership proof that involves showing a different leaf.
 			// Add the leaf data to the proof.
 			nonMembershipLeafData = leafData
 		}
 	}
 
-	proof := SparseMerkleProof{
+	return SparseMerkleProof{
```

Comment:
Is this change idiomatic Go? I prefer the old version personally since it's clearer what's being returned without having to parse two concepts at once. The compiler will optimize away the variable anyways, if performance was the concern.

Reply:
I got this and similar ones flagged by a linter... Apparently, I am using a more exhaustive linter, maybe we should agree on a linter config for all the repos, and respect all the errors it is showing.

Reply:
Hm. How about revert these changes for now, then we can open a separate issue (potentially also to be implemented across the org) on linting?

Reply:
sounds good. Should I create an issue ? or just leave it for now ?